src/injected/adapters/scan-results-to-unified-results.ts

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import { ResultStatus, UnifiedResult, UnifiedResults } from '../../common/types/store-data/unified-data-interface';
+import { AxeNodeResult, RuleResult, ScanResults } from '../../scanner/iruleresults';
+
+type UUIDGeneratorType = () => string;
+
+interface RuleResultData {
+    status: ResultStatus;
+    ruleID: string;
+}
+
+interface CreationData extends RuleResultData {
+    cssSelector: string;
+    snippet: string;
+    howToFix: {
+        oneOf: string[];
+        none: string[];
+        all: string[];
+    };
+}
+
+export const convertScanResultsToUnifiedResults = (scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResults => {
+    return {
+        results: createUnifiedResultsFromScanResults(scanResults, uuidGenerator),
+    } as UnifiedResults;
+};
+
+function createUnifiedResultsFromScanResults(scanResults: ScanResults, uuidGenerator: UUIDGeneratorType): UnifiedResult[] {
+    let resultsArray: UnifiedResult[] = [];
+
+    resultsArray = createUnifiedResultsFromRuleResults(scanResults.violations, 'fail', uuidGenerator);
```

Comment:
are we able to use spread syntax here to avoid mutating `resultsArray`? we might be able to return `[ ...createUnified(fail), ...createUnified(pass)]`